base/regex.jl

```diff
@@ -249,3 +249,19 @@ function eachmatch(re::Regex, str::AbstractString, ovr::Bool=false)
 end
 
 eachmatch(re::Regex, str::AbstractString) = RegexMatchIterator(re,str)
+
+## comparison ##
+
+function ==(a::Regex, b::Regex)
+    (a.pattern != b.pattern) && return false
+    (a.compile_options != b.compile_options) && return false
+    (a.match_options != b.match_options) && return false
+    return true
+end
+
+function isequal(a::Regex, b::Regex)
+    !isequal(a.pattern, b.pattern) && return false
+    !isequal(a.compile_options, b.compile_options) && return false
+    !isequal(a.match_options, b.match_options) && return false
+    return true
+end
```

Comment:
`==` and `isequal` do the same thing on all of these types, so the second definition is unnecessary.